File: snippets/curves/quadratic.py
import math


def evaluate_quadratic(x, a, b, c):
    """Return the polynomial value and its derivative at x."""
    if not all(math.isfinite(n) for n in [x, a, b, c]):
        raise ValueError("Use finite inputs.")
    value = (a * x + b) * x + c
    derivative = 2 * a * x + b
    if not all(math.isfinite(n) for n in [value, derivative]):
        raise ValueError("Result overflow.")
    return {"value": value, "derivative": derivative}

File: snippets/curves/quadratic.js
// Return the polynomial value and its derivative at x.
function evaluate_quadratic(x, a, b, c) {
  if (![x, a, b, c].every(Number.isFinite)) throw new RangeError("Use finite inputs.");
  const value = (a * x + b) * x + c;
  const derivative = 2 * a * x + b;
  if (![value, derivative].every(Number.isFinite)) throw new RangeError("Result overflow.");
  return { value, derivative };
}
